### ISSUE TITLE
feature request: warn on loss of error information

### ISSUE THREAD

**Issue by twpayne** [golang/lint#339]
I occasionally see code from Go newbies that discards error information, e.g.

```go
func foo(filename string) error {
    _, err := os.Open(filename)
    if err != nil {
        return errors.New("something broke")
    }
    // ...
}
```

In this example, the error returned by the function does not depend on the tested error, or any other parameter. The loss of error information makes debugging very hard as there is no indication of the underlying cause. This is particularly insidious in library code where such functions "dead end" diagnosis.

This is possibly something that `golint` could catch. The heuristic could be something like: "when returning an `error` in response to an `err != nil` test, the returned `error` should include either the error that was tested or some other variable".

**Comment by andybons** [comment]
Thanks for the issue.

This appears to be [out of scope](https://github.com/golang/lint/blob/master/README.md#scope) and so is being closed. If you feel this is in error please feel free to re-open.

Please note this is not the forum to discuss edits to [CodeReviewComments](https://golang.org/wiki/CodeReviewComments). If you wish to propose changes to it please send a mail to [golang-nuts](https://groups.google.com/forum/#!forum/golang-nuts).
